hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyReadWriteListOps.java

```diff
@@ -212,16 +230,22 @@ keyName, objectSizeInBytes, null, new HashMap())) {
     }
   }
 
-  public TaskType decideReadOrWriteTask() {
-    if (percentageRead == 100) {
-      return TaskType.READ_TASK;
-    } else if (percentageRead == 0) {
-      return TaskType.WRITE_TASK;
+  public void processListTasks(OzoneClient ozoneClient)
+      throws RuntimeException, IOException {
+    try {
+      ozoneClient.getProxy()
+          .listKeys(volumeName, bucketName, getPrefix(), null, maxListResult);
+    } catch (Exception ex) {
+      throw ex;
     }
-    //mix workload
+  }
+
+  public TaskType decideReadWriteOrListTask() {
     int tmp = ThreadLocalRandom.current().nextInt(1, 101);
-    if (tmp  <= percentageRead) {
+    if (tmp <= percentageRead) {
       return TaskType.READ_TASK;
+    } else if (tmp > percentageRead && tmp <= percentageRead + percentageList) {
```

Comment:
```suggestion
    } else if (tmp <= percentageRead + percentageList) {
```
redundant condition `tmp > percentageRead` as it's true already.

Reply:
good catch!